daemon/graphdriver/overlay2/overlay.go

```diff
@@ -15,6 +15,7 @@ import (
 	"strings"
 	"sync"
 
+	ctdoverlay "github.com/containerd/containerd/snapshots/overlay"
```

Comment:
RFC: do we want to import this to reduce code copypasta, or do we want to just copy the code without introducing dependency on gh/c/c/snapshots ?

Reply:
Hm... good one; "generally" I'd say "don't duplicate", but in this case; some thoughts;

- the code currently lives in containerd's overlay snapshotter code (which we _don't_ need)
  - _if_ we want to use the containerd code, I'd prefer it to be refactored to be in (e.g.) containerd/containerd/snapshots/overlay/utils` (or `overlayutils`)
  - in that case, we should also check what the overlap is between our "overlayutils" and the "containerd" one; for example, [`supportsMultipleLowerDir()`](https://github.com/containerd/containerd/blob/ddf6594fbeeddf30a55e0d2bcdb85956ec792fd2/snapshots/overlay/check.go#L34-L41) looks to be a fork of our `overlayutils` package
- (this depends a bit on how 'stable' that part of the code is, because the downside of depending on it from containerd means that if we need a fix in that code, we need to update the _whole_ containerd dependency (which could bring a lot of additional / updated dependencies due to how it's structured)

From the above, I'm tempted to say;

1. Start with copying the check to our own `graphdriver/overlayutils` package
2. After that (follow-up), refactor containerd to have it in a `(overlay)utils` package
3. Look at the overlap between our `overlayutils` and containerd's utils, and consider replacing our overlayutils with the one from containerd

Reply:
@cpuguy83 @tonistiigi WDYT?

Reply:
updated